test/e2e/nodepool_nto_machineconfig_test.go

```diff
@@ -46,222 +50,281 @@ spec:
+				Name:      "hugepages-tuned-test",
+				Namespace: hostedCluster.Namespace,
+			},
+			Data: map[string]string{tuningConfigKey: hugepagesTuned},
+		}
 
-	var nodePool hyperv1.NodePool
-	for _, nodePool = range nodePools.Items {
-		if nodePool.Spec.ClusterName != hostedCluster.Name {
-			continue
+		if err := mgmtClient.Create(ctx, tuningConfigConfigMap); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create configmap for custom Tuned object: %v", err)
+			}
 		}
 
 		np := nodePool.DeepCopy()
 		nodePool.Spec.TuningConfig = append(nodePool.Spec.TuningConfig, corev1.LocalObjectReference{Name: tuningConfigConfigMap.Name})
-		if err := client.Patch(ctx, &nodePool, crclient.MergeFrom(np)); err != nil {
+		if err := mgmtClient.Patch(ctx, nodePool, crclient.MergeFrom(np)); err != nil {
 			t.Fatalf("failed to update nodepool %s after adding Tuned config: %v", nodePool.Name, err)
 		}
-	}
 
-	ds := ntoMachineConfigUpdatedVerificationDS.DeepCopy()
-	if err := guestClient.Create(ctx, ds); err != nil {
-		t.Fatalf("failed to create %s DaemonSet in guestcluster: %v", ds.Name, err)
-	}
-
-	t.Logf("waiting for NodePools in-place update with NTO-generated MachineConfig")
-	err = wait.PollImmediateWithContext(ctx, 5*time.Second, 15*time.Minute, func(ctx context.Context) (bool, error) {
-		if ctx.Err() != nil {
-			return false, err
-		}
-		pods := &corev1.PodList{}
-		if err := guestClient.List(ctx, pods, crclient.InNamespace(ds.Namespace), crclient.MatchingLabels(ds.Spec.Selector.MatchLabels)); err != nil {
-			t.Logf("WARNING: failed to list pods, will retry: %v", err)
-			return false, nil
-		}
-		nodes := &corev1.NodeList{}
-		if err := guestClient.List(ctx, nodes); err != nil {
-			t.Logf("WARNING: failed to list nodes, will retry: %v", err)
-			return false, nil
-		}
-		if len(pods.Items) != len(nodes.Items) {
-			return false, nil
+		ds := ntoMachineConfigUpdatedVerificationDS.DeepCopy()
+		ds.Name = ds.Name + "-inplace"
+		if err := hostedClusterClient.Create(ctx, ds); err != nil {
+			if !errors.IsAlreadyExists(err) {
+				t.Fatalf("failed to create %s DaemonSet in guestcluster: %v", ds.Name, err)
+			}
 		}
 
-		for _, pod := range pods.Items {
-			if !isPodReady(&pod) {
+		t.Logf("waiting for NodePools in-place update with NTO-generated MachineConfig")
+		err = wait.PollImmediateWithContext(ctx, 5*time.Second, 15*time.Minute, func(ctx context.Context) (bool, error) {
+			if ctx.Err() != nil {
+				return false, err
+			}
+			pods := &corev1.PodList{}
+			if err := hostedClusterClient.List(ctx, pods, crclient.InNamespace(ds.Namespace), crclient.MatchingLabels(ds.Spec.Selector.MatchLabels)); err != nil {
+				t.Logf("WARNING: failed to list pods, will retry: %v", err)
 				return false, nil
 			}
-		}
 
-		return true, nil
-	})
-	if err != nil {
-		t.Fatalf("failed waiting for all pods in the NTO MachineConfig update verification DS to be ready: %v", err)
-	}
+			if len(pods.Items) < len(nodes) {
```

Comment:
why relaxing this?

Reply:
because they share the same HostedCluster and they have the same labels. For now this is temporary measure until I finish the PR (Still not there). I've submitted it to share thoughts with Cesar and you ;). The plan that I've though is to add a Label to identify it as unique and set the older one.

Reply:
We know which NodePool the test is operating against. We know which Nodes (they have a NodePool label) belong to a NodePool. So I'm not seeing the need to relax comparing to current approach.